File: nyancat/src/main/java/pyxis/uzuki/live/nyancat/NyanCat.java
package pyxis.uzuki.live.nyancat;

import java.util.List;

/**
 * NyanCat
 * Class: NyanCat
 * Created by Pyxis on 2017-10-30.
 * <p>
 * Description:
 */

@SuppressWarnings("UnusedDeclaration")
public final class NyanCat {

    private NyanCat() {

    }

    public static void d(String message) {
        NyanCatStatic.logger.d(message);
    }

    public static void d(String message, Object... args) {
        NyanCatStatic.logger.d(message, args);
    }

    public static void d(Throwable t, String message, Object... args) {
        NyanCatStatic.logger.d(t, message, args);
    }

    public static <T> void d(List<T> lists) {
        NyanCatStatic.logger.d(lists);
    }

    public static <T> void d(CharSequence delimiter, List<T> lists) {
        NyanCatStatic.logger.d(delimiter, lists);
    }

    public static void d(Object object) {
        NyanCatStatic.logger.d(object);
    }

    public static void i(String message) {
        NyanCatStatic.logger.i(message);
    }

    public static void i(String message, Object... args) {
        NyanCatStatic.logger.i(message, args);
    }

    public static void i(Throwable t, String message, Object... args) {
        NyanCatStatic.logger.i(t, message, args);
    }

    public static <T> void i(List<T> lists) {
        NyanCatStatic.logger.i(lists);
    }

    public static <T> void i(CharSequence delimiter, List<T> lists) {
        NyanCatStatic.logger.i(delimiter, lists);
    }

    public static void i(Object object) {
        NyanCatStatic.logger.i(object);
    }

    public static void w(String message) {
        NyanCatStatic.logger.w(message);
    }

    public static void w(String message, Object... args) {
        NyanCatStatic.logger.w(message, args);
    }

    public static void w(Throwable t, String message, Object... args) {
        NyanCatStatic.logger.w(t, message, args);
    }

    public static <T> void w(List<T> lists) {
        NyanCatStatic.logger.w(lists);
    }

    public static <T> void w(CharSequence delimiter, List<T> lists) {
        NyanCatStatic.logger.w(delimiter, lists);
    }

    public static void w(Object object) {
        NyanCatStatic.logger.w(object);
    }

    public static void w(Throwable t) {
        NyanCatStatic.logger.w(t);
    }

    public static void e(Throwable t) {
        NyanCatStatic.logger.e(t);
    }

    public static void e(String message) {
        NyanCatStatic.logger.e(message);
    }

    public static void e(String message, Object... args) {
        NyanCatStatic.logger.e(message, args);
    }

    public static void e(Throwable t, String message, Object... args) {
        NyanCatStatic.logger.e(t, message, args);
    }

    public static <T> void e(List<T> lists) {
        NyanCatStatic.logger.e(lists);
    }

    public static <T> void e(CharSequence delimiter, List<T> lists) {
        NyanCatStatic.logger.e(delimiter, lists);
    }

    public static void e(Object object) {
        NyanCatStatic.logger.e(object);
    }

    public static void v(String message) {
        NyanCatStatic.logger.v(message);
    }

    public static void v(String message, Object... args) {
        NyanCatStatic.logger.v(message, args);
    }

    public static void v(Throwable t, String message, Object... args) {
        NyanCatStatic.logger.v(t, message, args);
    }

    public static <T> void v(List<T> lists) {
        NyanCatStatic.logger.v(lists);
    }

    public static <T> void v(CharSequence delimiter, List<T> lists) {
        NyanCatStatic.logger.v(delimiter, lists);
    }

    public static void v(Object object) {
        NyanCatStatic.logger.v(object);
    }

    public static NyanCatLogger tag(String tag) {
        return NyanCatStatic.tag(tag);
    }
}

File: nyancat/src/main/java/pyxis/uzuki/live/nyancat/NyanCat.kt
package pyxis.uzuki.live.nyancat

import android.content.Intent
import android.os.Bundle

object NyanCat {

    @JvmStatic
    fun d(message: String) {
        NyanCatGlobal.logger.d(message)
    }

    @JvmStatic
    fun d(message: String, vararg args: Any) {
        NyanCatGlobal.logger.d(message, *args)
    }

    @JvmStatic
    fun d(t: Throwable, message: String, vararg args: Any) {
        NyanCatGlobal.logger.d(t, message, *args)
    }

    @JvmStatic
    fun <T> d(lists: List<T>) {
        NyanCatGlobal.logger.d(lists)
    }

    @JvmStatic
    fun <T> d(delimiter: CharSequence, lists: List<T>) {
        NyanCatGlobal.logger.d(delimiter, lists)
    }

    @JvmStatic
    fun d(`object`: Any) {
        NyanCatGlobal.logger.d(`object`)
    }

    @JvmStatic
    fun <K, V> d(map: Map<K, V>) {
        NyanCatGlobal.logger.d(map)
    }

    @JvmStatic
    fun <K, V> d(delimiter: CharSequence, map: Map<K, V>) {
        NyanCatGlobal.logger.d(delimiter, map)
    }

    @JvmStatic
    fun d(bundle: Bundle) {
        NyanCatGlobal.logger.d(bundle)
    }

    @JvmStatic
    fun d(delimiter: CharSequence, bundle: Bundle) {
        NyanCatGlobal.logger.d(delimiter, bundle)
    }

    @JvmStatic
    fun d(intent: Intent) {
        NyanCatGlobal.logger.d(intent)
    }

    @JvmStatic
    fun d(delimiter: CharSequence, intent: Intent) {
        NyanCatGlobal.logger.d(delimiter, intent)
    }

    @JvmStatic
    fun i(message: String) {
        NyanCatGlobal.logger.i(message)
    }

    @JvmStatic
    fun i(message: String, vararg args: Any) {
        NyanCatGlobal.logger.i(message, *args)
    }

    @JvmStatic
    fun i(t: Throwable, message: String, vararg args: Any) {
        NyanCatGlobal.logger.i(t, message, *args)
    }

    @JvmStatic
    fun <T> i(lists: List<T>) {
        NyanCatGlobal.logger.i(lists)
    }

    @JvmStatic
    fun <T> i(delimiter: CharSequence, lists: List<T>) {
        NyanCatGlobal.logger.i(delimiter, lists)
    }

    @JvmStatic
    fun i(`object`: Any) {
        NyanCatGlobal.logger.i(`object`)
    }

    @JvmStatic
    fun <K, V> i(map: Map<K, V>) {
        NyanCatGlobal.logger.i(map)
    }

    @JvmStatic
    fun <K, V> i(delimiter: CharSequence, map: Map<K, V>) {
        NyanCatGlobal.logger.i(delimiter, map)
    }

    @JvmStatic
    fun i(bundle: Bundle) {
        NyanCatGlobal.logger.i(bundle)
    }

    @JvmStatic
    fun i(delimiter: CharSequence, bundle: Bundle) {
        NyanCatGlobal.logger.i(delimiter, bundle)
    }

    @JvmStatic
    fun i(intent: Intent) {
        NyanCatGlobal.logger.i(intent)
    }

    @JvmStatic
    fun i(delimiter: CharSequence, intent: Intent) {
        NyanCatGlobal.logger.i(delimiter, intent)
    }

    @JvmStatic
    fun w(message: String) {
        NyanCatGlobal.logger.w(message)
    }

    @JvmStatic
    fun w(message: String, vararg args: Any) {
        NyanCatGlobal.logger.w(message, *args)
    }

    @JvmStatic
    fun w(t: Throwable, message: String, vararg args: Any) {
        NyanCatGlobal.logger.w(t, message, *args)
    }

    @JvmStatic
    fun <T> w(lists: List<T>) {
        NyanCatGlobal.logger.w(lists)
    }

    @JvmStatic
    fun <T> w(delimiter: CharSequence, lists: List<T>) {
        NyanCatGlobal.logger.w(delimiter, lists)
    }

    @JvmStatic
    fun w(`object`: Any) {
        NyanCatGlobal.logger.w(`object`)
    }

    @JvmStatic
    fun w(t: Throwable) {
        NyanCatGlobal.logger.w(t)
    }

    @JvmStatic
    fun <K, V> w(map: Map<K, V>) {
        NyanCatGlobal.logger.w(map)
    }

    @JvmStatic
    fun <K, V> w(delimiter: CharSequence, map: Map<K, V>) {
        NyanCatGlobal.logger.w(delimiter, map)
    }

    @JvmStatic
    fun w(bundle: Bundle) {
        NyanCatGlobal.logger.w(bundle)
    }

    @JvmStatic
    fun w(delimiter: CharSequence, bundle: Bundle) {
        NyanCatGlobal.logger.w(delimiter, bundle)
    }

    @JvmStatic
    fun w(intent: Intent) {
        NyanCatGlobal.logger.w(intent)
    }

    @JvmStatic
    fun w(delimiter: CharSequence, intent: Intent) {
        NyanCatGlobal.logger.w(delimiter, intent)
    }

    @JvmStatic
    fun e(t: Throwable) {
        NyanCatGlobal.logger.e(t)
    }

    @JvmStatic
    fun e(message: String) {
        NyanCatGlobal.logger.e(message)
    }

    @JvmStatic
    fun e(message: String, vararg args: Any) {
        NyanCatGlobal.logger.e(message, *args)
    }

    @JvmStatic
    fun e(t: Throwable, message: String, vararg args: Any) {
        NyanCatGlobal.logger.e(t, message, *args)
    }

    @JvmStatic
    fun <T> e(lists: List<T>) {
        NyanCatGlobal.logger.e(lists)
    }

    @JvmStatic
    fun <T> e(delimiter: CharSequence, lists: List<T>) {
        NyanCatGlobal.logger.e(delimiter, lists)
    }

    @JvmStatic
    fun e(`object`: Any) {
        NyanCatGlobal.logger.e(`object`)
    }

    @JvmStatic
    fun <K, V> e(map: Map<K, V>) {
        NyanCatGlobal.logger.e(map)
    }

    @JvmStatic
    fun <K, V> e(delimiter: CharSequence, map: Map<K, V>) {
        NyanCatGlobal.logger.e(delimiter, map)
    }

    @JvmStatic
    fun e(bundle: Bundle) {
        NyanCatGlobal.logger.e(bundle)
    }

    @JvmStatic
    fun e(delimiter: CharSequence, bundle: Bundle) {
        NyanCatGlobal.logger.e(delimiter, bundle)
    }

    @JvmStatic
    fun e(intent: Intent) {
        NyanCatGlobal.logger.e(intent)
    }

    @JvmStatic
    fun e(delimiter: CharSequence, intent: Intent) {
        NyanCatGlobal.logger.e(delimiter, intent)
    }

    @JvmStatic
    fun v(message: String) {
        NyanCatGlobal.logger.v(message)
    }

    @JvmStatic
    fun v(message: String, vararg args: Any) {
        NyanCatGlobal.logger.v(message, *args)
    }

    @JvmStatic
    fun v(t: Throwable, message: String, vararg args: Any) {
        NyanCatGlobal.logger.v(t, message, *args)
    }

    @JvmStatic
    fun <T> v(lists: List<T>) {
        NyanCatGlobal.logger.v(lists)
    }

    @JvmStatic
    fun <T> v(delimiter: CharSequence, lists: List<T>) {
        NyanCatGlobal.logger.v(delimiter, lists)
    }

    @JvmStatic
    fun v(`object`: Any) {
        NyanCatGlobal.logger.v(`object`)
    }

    @JvmStatic
    fun <K, V> v(map: Map<K, V>) {
        NyanCatGlobal.logger.v(map)
    }

    @JvmStatic
    fun <K, V> v(delimiter: CharSequence, map: Map<K, V>) {
        NyanCatGlobal.logger.v(delimiter, map)
    }

    @JvmStatic
    fun v(bundle: Bundle) {
        NyanCatGlobal.logger.v(bundle)
    }

    @JvmStatic
    fun v(delimiter: CharSequence, bundle: Bundle) {
        NyanCatGlobal.logger.v(delimiter, bundle)
    }

    @JvmStatic
    fun v(intent: Intent) {
        NyanCatGlobal.logger.v(intent)
    }

    @JvmStatic
    fun v(delimiter: CharSequence, intent: Intent) {
        NyanCatGlobal.logger.v(delimiter, intent)
    }

    @JvmStatic
    fun tag(tag: String): NyanCatLogger {
        return NyanCatGlobal.tag(tag)
    }
}